src/utils/segmentInitializer.ts

```diff
@@ -0,0 +1,47 @@
+import { Logger } from './logger';
+import Analytics from 'analytics-node';
+
+export namespace SegmentInitializer {
+  export function initialize(
+    clientSegmentKey: string | undefined = undefined
+  ): Analytics | undefined {
+    const segmentWriteKey: string | undefined =
+      clientSegmentKey || getSegmentWriteKey();
+
+    if (segmentWriteKey) {
+      /* 
+        flushAt: Number ->  The number of messages to enqueue before flushing.
+        flushInterval: Number ->    The number of milliseconds to wait before flushing the queue automatically.
+        ref: https://segment.com/docs/connections/sources/catalog/libraries/server/node/#configuration
+        */
+      let analytics: Analytics = new Analytics(segmentWriteKey, {
+        flushAt: 1,
+        flushInterval: 10000,
+      });
+      return analytics;
+    } else {
+      Logger.log(
+        'Missing segmentWriteKey from package.json OR package.json in vscode-commons'
+      );
+      return undefined;
+    }
+  }
+}
+
+function getSegmentWriteKey(): string | undefined {
```

Comment:
Should this be called getDefaultSegmentWriteKey()?

Reply:
done